packages/serverpod_shared/test/session_logs_config_test.dart

```diff
@@ -0,0 +1,186 @@
+      );
+
+      expect(config.sessionLogs, isNull);
+    },
+  );
+
+  test(
+    'Given a Serverpod config with sessionLogs and no database when persistentEnabled is true, it is overridden to false.',
+    () {
+      var serverpodConfig = '''
+apiServer:
+  port: 8080
+  publicHost: localhost
+  publicPort: 8080
+  publicScheme: http
+sessionLogs:
+  persistentEnabled: true
+  consoleEnabled: false
+''';
+
+      var config = ServerpodConfig.loadFromMap(
+        runMode,
+        serverId,
+        passwords,
+        loadYaml(serverpodConfig),
+        environment: {},
+      );
+
+      expect(config.sessionLogs?.persistentEnabled, isFalse);
+      expect(config.sessionLogs?.consoleEnabled, isFalse);
+    },
+  );
+
+  test(
+    'Given a Serverpod config with sessionLogs and no database when persistentEnabled is false, it remains false.',
+    () {
+      var serverpodConfig = '''
+apiServer:
+  port: 8080
+  publicHost: localhost
+  publicPort: 8080
+  publicScheme: http
+sessionLogs:
+  persistentEnabled: false
+  consoleEnabled: true
+''';
+
+      var config = ServerpodConfig.loadFromMap(
+        runMode,
+        serverId,
+        passwords,
+        loadYaml(serverpodConfig),
+        environment: {},
+      );
+
+      expect(config.sessionLogs?.persistentEnabled, isFalse);
+      expect(config.sessionLogs?.consoleEnabled, isTrue);
+    },
+  );
+
+  test(
+    'Given a Serverpod config with sessionLogs and database when persistentEnabled is true, persistentEnabled remains true.',
+    () {
+      var serverpodConfig = '''
+apiServer:
+  port: 8080
+  publicHost: localhost
+  publicPort: 8080
+  publicScheme: http
+database:
+  host: localhost
+  port: 5432
+  name: testDb
+  user: testUser
+sessionLogs:
+  persistentEnabled: true
+  consoleEnabled: false
+''';
+
+      var config = ServerpodConfig.loadFromMap(
+        runMode,
+        serverId,
+        passwords,
+        loadYaml(serverpodConfig),
+      );
+
+      expect(config.sessionLogs?.persistentEnabled, isTrue);
+      expect(config.sessionLogs?.consoleEnabled, isFalse);
+    },
+  );
+
+  test(
+    'Given a Serverpod config with sessionLogs and database when persistentEnabled is false, persistentEnabled remains false.',
+    () {
+      var serverpodConfig = '''
+apiServer:
+  port: 8080
+  publicHost: localhost
+  publicPort: 8080
+  publicScheme: http
+database:
+  host: localhost
+  port: 5432
+  name: testDb
+  user: testUser
+sessionLogs:
+  persistentEnabled: false
+  consoleEnabled: true
+''';
+
+      var config = ServerpodConfig.loadFromMap(
+        runMode,
+        serverId,
+        passwords,
+        loadYaml(serverpodConfig),
+      );
+
+      expect(config.sessionLogs?.persistentEnabled, isFalse);
+      expect(config.sessionLogs?.consoleEnabled, isTrue);
+    },
+  );
+
+  test(
+    'Given a Serverpod config with sessionLogs and database when environment variables override them, sessionLogs config reflects the environment overrides.',
```

Comment:
Suggestion: use "then" instead of comma, applies to whole file
```suggestion
    'Given a Serverpod config with sessionLogs and database when environment variables override them then sessionLogs config reflects the environment overrides.',
```